### PR TITLE
Feature: Highlight trailing whitespace in Pretty formats

### LINKED ISSUE #71590: Highlight trailing whitespaces in Pretty formats
**Use case**

Make trailing whitespaces visible.

**Describe the solution you'd like**

If colors are enabled, make them underscored and set the color to gray.
This way they will still be copy-pasteable as whitespaces, but highlighted in the terminal.

### PR DESCRIPTION
**Pull request by bradenburns**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add highlighting and underscoring to trailing whitespace in Pretty formats when colors is enabled - enabled with `output_format_pretty_highlight_trailing_whitespace` setting, which is false by default

Closes #71590 

<!---
Directly edit documentation source files in the "docs" folder with the same pull-request as code changes

or

Add a user-readable short description of the changes that should be added to docs.clickhouse.com below.

### CODE REVIEW

File: preprocessed_configs/config.xml
Comment (alexkats) [inline preprocessed_configs/config.xml:1]: Please, remove this file from changelist

File: contrib/NuRaft
Comment (alexkats) [inline contrib/NuRaft:1]: What is the reason for updating this and other submodules (especially in this PR)?

**Review by alexkats — CHANGES_REQUESTED** [review]
Thanks for the changes, from the first look the idea looks good, but there are some unnecessary changes here

**Comment by alexkats** [comment]
I also believe that you used 2 different emails for commits, and one of those isn't tied to the GitHub account and as a result the CLA check has failed

**Comment by alexey-milovidov** [comment]
Thanks, this is a great work! The PR contains a few minor problems with the code - I will resubmit it as another PR and acknowledge your authorship.

**Comment by alexey-milovidov** [comment]
#73847

**Pull request closed without merging**